### PR TITLE
feat: button modifications

### PR DESCRIPTION
**Pull request by EthanStandel**
feat: button modifications, uses functional component styling, extends native props, a11y fix, allows anchor buttons, utilizes forwardRef

### CODE REVIEW

File: src/components/Button/ButtonContent.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      return (
+        <>
+          <span className='spinner--overlay'>
+            <Spinner intent={intent || 'default'} size={size} />
+          </span>
+          <span className='visibility--hidden'>{iconizedChildren}</span>
+        </>
+      );
+    } else {
+      return iconizedChildren;
```
Comment (kysley) [inline src/components/Button/ButtonContent.tsx:30]: Same as above, no `else` after returning

**Comment by kysley** [comment]
Looks good! I was actually working on this exact thing, but slightly differently :) I had adopted [https://stitches.dev](stitches) to handle theming, variants, design tokens (which look like they could use some standardization 😅) etc. 

If this gets approved I can propose that idea in a separate pr 🚀

File: src/components/Button/Button.tsx [hunk 2]
```diff
@@ -1,23 +1,29 @@
...
+      disabled: disabled || loading,
+      children: <ButtonContent {...props}>{children}</ButtonContent>,
+      ...otherNativeProps
+    };
+    
+    if ("href" in props) {
+      return (
+        <a {...targetProps as NativeAnchorProps} />
+      );
+    } else {
```
[the remark this answers is not in the document]
Comment (EthanStandel) [reply src/components/Button/Button.tsx:23]: Imo it's slightly more readable to have the else statement 🤷 But you're def right that it's strictly for aesthetics

**Comment by EthanStandel** [comment]
@kysley because this only hits one component, I'm kind of just leaving it around as recommendations @joshdschneider 

I'm not sure these changes will actually get merged as is, and I'm cool if they either get ditched/cherry-picked or merged

File: src/components/Button/Button.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (kysley) [reply src/components/Button/Button.tsx:23]: Completely agree, perhaps we should add prettier & eslint

[the remark this answers is not in the document]
Comment (EthanStandel) [reply src/components/Button/Button.tsx:23]: That's always a fair recommendation for any open source project

**Comment by joshdschneider** [comment]
@kysley this is my first component library (I'm still learning 😅), but @EthanStandel has been super helpful showing me the ropes. Don't think I'll merge this pr, was going to start with passing forward all the native props for each component. Then merge that and update the docs.

Also, [stitches](https://stitches.dev/) looks legit! Was going to try out [emotion](https://emotion.sh/) to clean up the scss but I'm down to give stitches a go. Feel free to contribute if you want, could use all the help I can get. I just added prettier and eslint.

**Comment by kysley** [comment]
@joshdschneider Not a problem 😄 I'll gladly contribute where you would like me too. This is a beautiful component library you have ✨ 

I'll see if I can get us started w/ stitches

**Comment by joshdschneider** [comment]
Thanks @kysley! Have at it!

**Pull request closed without merging**